### PR TITLE
Enable users to add exclusion date, #5

### PR DESCRIPTION
**Pull request by indirap**
I wasn't sure where you wanted to add the feature, so I added it on the main page, with an `input` and `DatePicker`. Let me know what you think, @michaelghinrichs !
screen shot 2017-10-15 at 14 06 42

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,12 +17,27 @@ class App extends Component {
       resultDays: 0,
       calculatorInfo: this.decodeToState(),
       // uncomment this if you don't want to have to read from an encoded URL
-      // calculatorInfo: excludeDates,
+      calculatorInfo: excludedDates,
```
Comment (indirap) [inline src/App.js:20]: I'll comment this back out before merging, but also wanted to point out that there was a typo in the variable 🙂

File: src/App.js [hunk 2]
```diff
@@ -129,6 +171,28 @@ class App extends Component {
           <div className="Title">
             { calculatorInfo && calculatorInfo.title}
           </div>
+          <div className="Content Stack">
```
Comment (indirap) [inline src/App.js:174]: I'm not great at styling front-end stuff, so I just used the same class names as what you had for calculating exclusion dates

**Comment by hanneswidrig** [comment]
looks great to me, I will let michaelghinrichs merge this since there is a bit more going on.

File: src/App.js (hunk 1 above)
Comment (karismatic-megafauna) [reply src/App.js:20]: thank you!

File: src/App.js [hunk 3]
```diff
@@ -69,6 +84,33 @@ class App extends Component {
...
+      return;
+    }
+
+    if (!this.state.newExclusionReason) {
+      return;
+    }
+
+    let newExcludedDatesData = JSON.parse(JSON.stringify(this.state.calculatorInfo.data));
+    const newExcludedDate = {
+      "date": this.state.newExclusionDate.format("MM/DD/YYYY"),
```
Comment (karismatic-megafauna) [inline src/App.js:98]: non-blocking:

you don't need quotes around the keys in this object as these are valid identifier names :)

File: src/App.js [hunk 4]
```diff
@@ -129,6 +171,28 @@ class App extends Component {
...
             { calculatorInfo && calculatorInfo.title}
           </div>
+          <div className="Content Stack">
+            <div className="Control">
+              <input 
+                type="text" 
+                value={this.state.newExclusionReason}
+                onChange={this.handleReasonChange}
+                placeholder="Enter a date exclusion reason">
+              </input>
```
Comment (karismatic-megafauna) [inline src/App.js:181]: non-blocking: 
this tag can be self-closing, `/>` which would mean you can git rid of the `>` on 180

File: src/App.js [hunk 5]
```diff
@@ -129,6 +171,28 @@ class App extends Component {
...
+              &nbsp;&nbsp;
```
Comment (karismatic-megafauna) [inline src/App.js:182]: This is being used to get the spacing between the `input` and the `DatePicker`?

Maybe this is an opportunity to give it a className with some styles or to leverage some `flexbox` property...

**Review by karismatic-megafauna — APPROVED** [review]
Awesome work!

Thanks so much for contributing :) 

I was imagining this would be in the sidebar but this is in the right direction!

File: src/App.js (hunk 5 above)
Comment (indirap) [reply src/App.js:182]: Yeah, sorry, I'm really bad at CSS!

**Pull request merged**